Add JSON Parsing Capabilities (Press Contact!)

lolfuljames opened the pull request:
http://jtan-sps-summer20.appspot.com/
> press contact!
> see the arraylist being passed as JSON then parsed as a list

lolfuljames committed:
Add JSON Parsing Capabilities

henryken commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +25,18 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  ArrayList<String> randomMessages = new ArrayList<String>(Arrays.asList("This isn't it.", "Try harder bro.", "You should reconsider your choices.", "You did it! Here's a cookie."));
```
In OOP, it's a good practice to assign a type to its interface, unless there's a valid reason to refer to the subtype.
```java
List<String> randomMessages = new ArrayList<>(........);
```

This code can even be simplified to be
```java
List<String> randomMessages = Arrays.asList("...........");
```

lolfuljames replied:
Thanks for pointing it out! It's a first for me.

henryken replied:
That's alright!

henryken commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +25,18 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  ArrayList<String> randomMessages = new ArrayList<String>(Arrays.asList("This isn't it.", "Try harder bro.", "You should reconsider your choices.", "You did it! Here's a cookie."));
```
Try to adhere to the column limit coding style so that we avoid having to scroll horizontally if possible.
https://google.github.io/styleguide/javaguide.html#s4.4-column-limit

henryken commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -23,10 +25,18 @@
+    response.getWriter().println(data);
+  }
+
+  private String convertToJsonUsingGson(ArrayList<String> data) {
```
Pass the variable using the interface, i.e. List<String>

henryken reviewed (changes requested):
I have few comments.

lolfuljames committed:
Replace ArrayList with List

lolfuljames committed:
Fix ul buttons bullet position

This commit enables the bullet of the ul element within hello-container to be positioned
with the content.

henryken reviewed (approved):
LGTM
